client/src/controllers/LoaderController.ts

```diff
@@ -0,0 +1,121 @@
+import { Application, Controller } from '@hotwired/stimulus';
+import { gettext } from '../utils/gettext';
+
+/**
+ * <button
+ *   type="submit"
+ *   class="button button-longrunning"
+ *   data-controller="w-loader"
+ *   data-w-loader-active-class="button-longrunning-active"
+ *   data-w-loader-active-value="{% trans 'Signing in…' %}"
+ *   data-w-loader-duration-seconds-value="40"
+ *   data-action="w-loader#activate"
+ * >
+ *  {% icon name="spinner" %}
+ *  <em data-w-loader-target="label">{% trans 'Sign in' %}</em>
+ * </button>
+ */
+export class LoaderController extends Controller {
+  static classes = ['active'];
+  static targets = ['label'];
+  static values = {
+    active: { type: String, default: gettext('Loading') },
+    durationSeconds: { type: Number, default: 30 },
+  };
+
+  declare activeClass: string;
+  declare activeValue: string;
+  declare durationSecondsValue: number;
+  declare labelTarget: HTMLElement;
+  declare hasActiveClass: boolean;
+  declare hasLabelTarget: boolean;
+  declare originalText: string;
+
+  static afterLoad(identifier: string, application: Application) {
+    const { controllerAttribute } = application.schema;
+    const { actionAttribute } = application.schema;
+
+    const updateLoader = () => {
+      const em = document.querySelector('em') as HTMLElement;
+      document
+        .querySelectorAll(
+          `.button-longrunning:not([data-controller~='${identifier}'])`,
+        )
+        .forEach((button) => {
+          button.setAttribute(controllerAttribute, identifier);
+          button.setAttribute(actionAttribute, 'w-loader#activate');
+
+          if (button.getAttribute('data-clicked-text')) {
+            button.setAttribute(
+              'data-w-loader-active-value',
+              `${button.getAttribute('data-clicked-text')}`,
+            );
+            button.removeAttribute('data-clicked-text');
+          }
+
+          if (em.matches('button em')) {
+            em.setAttribute('data-w-loader-target', 'label');
+          }
+
+          button.setAttribute('data-w-loader-duration-seconds-value', '10');
+        });
+    };
+
+    document.addEventListener('DOMContentLoaded', updateLoader);
+  }
+
+  activate() {
+    this.originalText = this.hasLabelTarget
```

Comment:
```suggestion
    this.labelValue = this.hasLabelTarget
```
... instead of a non Stimulus value for the originalText, we can now use the `label` value.